### ISSUE TITLE
NULL variable replacement

### ISSUE THREAD

**Issue by ToMs77130** [grafana/grafana#96140]
### What happened?

I've got a dashboard variable which depends of another one, its SQL code is  

```
select folder ->>5 as res
FROM products
where $__timeFilter(time)
and ...
and (
      case 
        when folder->>4 is null then true
        else  folder->>4 in ($folder4)
      end
      )
```

This variable folder5 depends on folder4, which can sometimes be NULL.  (folder is a JSON field with variable depth).

In this example, this is the case, and it seems grafana doesn't want to replace $folder4 by a NULL value as it did before.
If I manually replace $folder4 by `''` , it works.

With 11.2 version this was working well, and since I've upgraded to 11.3 it does not works anymore, I've got following message in console and no requests are sent to database.

```
backend_srv.ts:127 PanelQueryRunner Error TypeError: Cannot read properties of null (reading 'replace')
    at sR.quoteLiteral (PostgresQueryModel.ts:22:24)
    at eval (SqlDatasource.ts:92:66)
    at Array.map (<anonymous>)
    at E_.interpolateVariable (SqlDatasource.ts:92:34)
    at so (sceneInterpolator.js:59:12)
    at sceneInterpolator.js:24:19
    at String.replace (<anonymous>)
    at ri (sceneInterpolator.js:14:17)
    at Object.Dh [as interpolate] (sceneGraph.js:31:10)
    at te.replace (template_srv.ts:264:25)
```

### What did you expect to happen?

Keep previous behavior with replacement of NULL variables by `''`

### Did this work before?

Yes, in 11.2

### How do we reproduce it?

1. Define variable var as `select NULL;`
2. Use variable in a panel SQL request, like `select xxx from yyy where var in ($var)`

### Environment (with versions)?

Grafana: 11.3
OS: Ubuntu 20.4
Browser: Chrome 129

### Grafana platform?

A package manager (APT, YUM, BREW, etc.)

**Comment by neshorg** [comment]
I just want to add that I've just now upgraded to `11.3.0+security-01` and got the same errors for multiple template variables being dependent on other template variables while using mysql. 

For me I'm using variables which should return a whole list of row values, but instead they're just returning the 1st value. 
Then again, I had to wrap my variable in single quotes before it even started to return the 1st value. However I'm stuck as to getting all values from the variable like before in an array-like form. [Formatting the value](https://grafana.com/docs/grafana/latest/dashboards/variables/variable-syntax/#csv) for example like csv doesn't help either.  

@ToMs77130 did you try wrapping ($folder4) in single quotes?
```
select folder ->>5 as res
FROM products
where $__timeFilter(time)
and ...
and (
      case 
        when folder->>4 is null then true
        else  folder->>4 in ('$folder4')
      end
      )
```
